covid19-dashboard/dc-covid19/src/App.tsx

```diff
@@ -15,185 +15,137 @@
-
-        Promise.all(requests).then(responses => {
-            responses.forEach((response, index) => {
-                response.json().then( json => {
-                    // Keep the data stored in the same position, that is: [cases, deaths, population, names]
-                    // Make a deep copy of the data stored in this.state.data, and then store the json at the index
-                    const currentData = this.state.allData.map(val => val)
-                    currentData[index] = json
-                    this.setState({allData: currentData})
-
-                    // Index 0 contains the 'total-cases' object, which has the dates as keys.
-                    if (index === 0) {
-                        // Get an array of all the dates in the data.
-                        const allDates = Object.keys(json)
-                        const latestDate = allDates[allDates.length - 1]
-                        // Get the deltaDates from the Configuration file.
-                        const datesToPick = getRealISODatesFromArrayOfDeltaDays(latestDate, Configuration.DATES)
-                        this.setState({datesToPick: datesToPick})
-                        // The default date is always the latest date available in the dataset,
-                        // which happens to be the last date in the array
-                        const mostRecentDate = datesToPick[datesToPick.length - 1]
-                        this.setState({datePicked: mostRecentDate})
-                    // Index 3 contains the 'geoId -> belongsToRegion' object
-                    // We can get all the available States
-                    } else if (index === 3){
-                        let regions: {} = {};
-                        Object.keys(json).forEach(geoId => {
-                            if (json[geoId][1] === 'country/USA') regions[geoId] = json[geoId][0]
-                        })
-                        this.setState({availableRegions: {...this.state.availableRegions, ...regions}})
-                    }
-                })
-            })
-        })
-    }
-
-    /**
-     * Creates a reference for each row which allows easy access from the SideNav.
-     */
-    generateReferences = () => {
-        let refs_ = {}
-        for (let key in ContentFile) refs_[key] = React.createRef<HTMLDivElement>()
-        return refs_
-    }
-
-    /**
-     * In charge of scrolling to a specific reference in the page.
-     * @param event
-     */
-    handleScrollOnRef = (event) => {
-        const ref_ = this.refs_[event.target.id]
-        if (ref_.current) {
-            ref_.current.scrollIntoView({
-                behavior: "smooth",
-                block: "nearest"
+  allData: {}[]; // [cases, deaths, population, names]
+  selectedRegion: string; // current region selected
+  selectedShowTopN: number; // default number of top counties/states to show
+  datePicked: string; // current date selected, latest is always default
```

Comment:
should datepicked and datesToPick be an IsoDate object/class or a string in ISO format?